pkg/encryption/cipher_test.go

```diff
@@ -48,15 +32,199 @@ func TestEncodeAndDecodeAccessTokenB64(t *testing.T) {
+				secret := make([]byte, secretSize)
+				_, err := io.ReadFull(rand.Reader, secret)
+				assert.Equal(t, nil, err)
+
+				c, err := NewBase64Cipher(initCipher, secret)
+				assert.Equal(t, nil, err)
+
+				// Test various sizes sessions might be
+				for _, dataSize := range []int{10, 100, 1000, 5000, 10000} {
+					data := make([]byte, dataSize)
+					_, err := io.ReadFull(rand.Reader, data)
+					assert.Equal(t, nil, err)
+
+					encrypted, err := c.Encrypt(data)
+					assert.Equal(t, nil, err)
+					assert.NotEqual(t, encrypted, data)
+
+					decrypted, err := c.Decrypt(encrypted)
+					assert.Equal(t, nil, err)
+					assert.Equal(t, data, decrypted)
+					assert.NotEqual(t, encrypted, decrypted)
+				}
+			})
+		}
+	}
+}
+
+func TestDecryptCFBWrongSecret(t *testing.T) {
+	secret1 := []byte("0123456789abcdefghijklmnopqrstuv")
+	secret2 := []byte("9876543210abcdefghijklmnopqrstuv")
+
+	c1, err := NewCFBCipher(secret1)
+	assert.Equal(t, nil, err)
+
+	c2, err := NewCFBCipher(secret2)
+	assert.Equal(t, nil, err)
+
+	data := []byte("f3928pufm982374dj02y485dsl34890u2t9nd4028s94dm58y2394087dhmsyt29h8df")
+
+	ciphertext, err := c1.Encrypt(data)
+	assert.Equal(t, nil, err)
+
+	wrongData, err := c2.Decrypt(ciphertext)
+	assert.Equal(t, nil, err)
+	assert.NotEqual(t, data, wrongData)
+}
+
+func TestDecryptGCMWrongSecret(t *testing.T) {
+	secret1 := []byte("0123456789abcdefghijklmnopqrstuv")
+	secret2 := []byte("9876543210abcdefghijklmnopqrstuv")
+
+	c1, err := NewGCMCipher(secret1)
+	assert.Equal(t, nil, err)
+
+	c2, err := NewGCMCipher(secret2)
+	assert.Equal(t, nil, err)
+
+	data := []byte("f3928pufm982374dj02y485dsl34890u2t9nd4028s94dm58y2394087dhmsyt29h8df")
+
+	ciphertext, err := c1.Encrypt(data)
+	assert.Equal(t, nil, err)
+
+	// GCM is authenticated - this should lead to message authentication failed
+	_, err = c2.Decrypt(ciphertext)
+	assert.Error(t, err)
+}
+
+func TestIntermixCiphersErrors(t *testing.T) {
+	// Encrypt with GCM, Decrypt with CFB: Results in Garbage data
+	// Test all 3 valid AES sizes
+	for _, secretSize := range []int{16, 24, 32} {
+		subTestName := fmt.Sprintf("GCM->CFB::%d", secretSize)
+		t.Run(subTestName, func(t *testing.T) {
+			secret := make([]byte, secretSize)
+			_, err := io.ReadFull(rand.Reader, secret)
+			assert.Equal(t, nil, err)
+
+			gcm, err := NewGCMCipher(secret)
+			assert.Equal(t, nil, err)
+
+			cfb, err := NewCFBCipher(secret)
+			assert.Equal(t, nil, err)
+
+			// Test various sizes sessions might be
+			for _, dataSize := range []int{10, 100, 1000, 5000, 10000} {
+				data := make([]byte, dataSize)
+				_, err := io.ReadFull(rand.Reader, data)
+				assert.Equal(t, nil, err)
+
+				encrypted, err := gcm.Encrypt(data)
+				assert.Equal(t, nil, err)
+				assert.NotEqual(t, encrypted, data)
+
+				decrypted, err := cfb.Decrypt(encrypted)
+				assert.Equal(t, nil, err)
+				// Data is mangled
+				assert.NotEqual(t, data, decrypted)
+				assert.NotEqual(t, encrypted, decrypted)
+			}
```

Comment:
Is this loop repeated several times in this file? Could we extract it to a function maybe?

Reply:
I think the error scenario tests are different than the standard Encrypt/Decrypt tests.  I merged the Standard vs Base64 wrapped tests into one, so that removed the duplication those two had.